toucan_connectors/postgres/utils.py

```diff
@@ -603,18 +603,32 @@
 }
 
 
-def _build_materialized_views_info_extraction_query(db_name: str | None) -> str:
+def _build_materialized_views_info_extraction_query(
+    *, db_name: str | None, schema_name: str | None, table_name: str | None, exclude_columns: bool
+) -> str:
     # Here, we need to query pg_catalog because materialized views are not a standard SQL feature
     # and are thus not available in information_schema.
     # The WHERE condition filters to retrieve materialized views only and excludes system columns
     # (see https://www.postgresql.org/docs/current/catalog-pg-attribute.html).
     # The CASE statement is to format postgres types to sql types as found in information_schema
     database_name = f"'{db_name}'" if db_name else "NULL"
-    return f"""
+    where_clause = "WHERE cls.relname in (SELECT matviewname FROM pg_matviews) AND attr.attnum >= 0\n"
+    if schema_name is not None:
+        where_clause += f"AND ns.nspname = '{schema_name}'\n"
+    if table_name is not None:
+        where_clause += f"AND cls.relname = '{table_name}'\n"
+    return (
+        f"""
     SELECT {database_name} AS "database",
     ns.nspname AS "schema",
     'view' AS "table_type",
-    cls.relname AS table_name,
+    cls.relname AS table_name,"""
+        + (
+            # If we want to exclude the columns, we do not join on the pg_catalog.pg_type
+            # `information_schema.columns` table, and simply return an empty array
```

Comment:
On peut pas juste ne pas avoir de champ colonnes du tout  ou à None ?
Soit directement en sql soit après en python

Reply:
Discussed together: We'll keep empty columns to not make the change breaking